src/ImageSharp/Formats/Png/PngEncoderOptionsHelpers.cs

```diff
@@ -148,5 +151,71 @@ private static int CalculateBytesPerPixel(PngColorType? pngColorType, bool use16
                     return use16Bit ? 8 : 4;
             }
         }
+
+        /// <summary>
+        /// Comes up with the appropriate PngColorType for some kinds of
+        /// IPixel. This is not exhaustive because not all options have
+        /// reasonable defaults
+        /// </summary>
+        private static PngColorType? SuggestColorType<TPixel>()
+        where TPixel : struct, IPixel<TPixel>
+        {
+            Type tPixel = typeof(TPixel);
+
+            if (tPixel == typeof(Alpha8))
+            {
+                return PngColorType.GrayscaleWithAlpha;
+            }
+            if (tPixel == typeof(Argb32))
+            {
+                return PngColorType.RgbWithAlpha;
+            }
+            if (tPixel == typeof(Rgb24))
+            {
+                return PngColorType.Rgb;
+            }
+            if (tPixel == typeof(Gray16))
+            {
+                return PngColorType.Grayscale;
+            }
+            if (tPixel == typeof(Gray8))
+            {
+                return PngColorType.Grayscale;
+            }
```

Comment:
@JimBobSquarePants we can get rid of these if-s later, by extending `PixelTypeInfo`, I have an idea for that.

Reply:
C# 8.0 switch expressions would also help a lot because they don't require compile time constants if I remember correctly.

Reply:
I've just picked up this PR and will continue the changes.